### PR TITLE
[util] Support cairo script as output format

### PR DESCRIPTION
**Pull request by khaledhosny**
Can be handy for debugging the series of operations cairo performs.

### CODE REVIEW

File: util/helper-cairo.hh [hunk 1]
```diff
@@ -67,6 +73,21 @@ _cairo_eps_surface_create_for_stream (cairo_write_func_t  write_func,
...
+
+static cairo_surface_t *
+_cairo_script_surface_create_for_stream (cairo_write_func_t write_func,
+				         void *closure,
+				         double width,
+				         double height,
+				         cairo_content_t content,
+				         image_protocol_t protocol HB_UNUSED)
+{
+  cairo_device_t *script = cairo_script_create_for_stream (write_func, closure);
```
Comment (khaledhosny) [inline util/helper-cairo.hh:87]: We are leaking the cairo device. No idea where to destroy it, though.

Comment (behdad) [reply util/helper-cairo.hh:87]: Attach it to the surface as user-data. See `png_closure`.

Comment (behdad) [reply util/helper-cairo.hh:87]: I'm on it.

Comment (khaledhosny) [reply util/helper-cairo.hh]: I just pushed the change.

Comment (behdad) [reply util/helper-cairo.hh]: Actually, just destroying script after creating the surface should do it, no?

Comment (khaledhosny) [reply util/helper-cairo.hh]: Right!

**Comment by behdad** [comment]
This one is probably a cairo bug?
```
==853969== 8,032 bytes in 7 blocks are definitely lost in loss record 18 of 19
==853969==    at 0x484280F: malloc (vg_replace_malloc.c:442)
==853969==    by 0x4AE2574: UnknownInlinedFun (cairo-malloc-private.h:174)
==853969==    by 0x4AE2574: _cairo_path_buf_create (cairo-path-fixed.c:768)
==853969==    by 0x4AE448C: _cairo_path_fixed_init_copy (cairo-path-fixed.c:150)
==853969==    by 0x4B69A27: _emit_path.lto_priv.0 (cairo-script-surface.c:1957)
==853969==    by 0x4B776EB: _cairo_script_surface_fill (cairo-script-surface.c:2749)
==853969==    by 0x4B19028: _cairo_surface_fill (cairo-surface.c:2453)
==853969==    by 0x4B197F4: _cairo_surface_wrapper_fill (cairo-surface-wrapper.c:397)
==853969==    by 0x4B01AEE: _cairo_recording_surface_replay_internal.lto_priv.0 (cairo-recording-surface.c:2257)
==853969==    by 0x4B77CB4: UnknownInlinedFun (cairo-recording-surface.c:2473)
==853969==    by 0x4B77CB4: _emit_scaled_glyph_vector (cairo-script-surface.c:3136)
==853969==    by 0x4B7826A: UnknownInlinedFun (cairo-script-surface.c:3259)
==853969==    by 0x4B7826A: _cairo_script_surface_show_text_glyphs (cairo-script-surface.c:3417)
==853969==    by 0x4B1C43C: _cairo_surface_show_text_glyphs (cairo-surface.c:2981)
==853969==    by 0x4AD5E74: _cairo_gstate_show_text_glyphs (cairo-gstate.c:2082)
```

File: util/helper-cairo.hh (hunk 1 above)
Comment (behdad) [reply util/helper-cairo.hh]: Are you pushing the simpler version?

**Comment by behdad** [comment]
LGTM either way.

**Pull request merged**